pydbtools/utils.py

```diff
@@ -1,18 +1,144 @@
+    has had __temp__ wrapped in quote marks.
+
+    Args:
+        sql_query (str): an SQL query
+
+    Raises:
+        ValueError
+    """
+    if re.findall(r'["|\']__temp__["|\']\.', sql_query.lower()):
+        raise ValueError(
+            "When querying a temporary database, __temp__ should not be wrapped in quotes"
+        )
+
+
+def clean_query(sql_query: str) -> str:
+    """
+    removes trailing whitespace, newlines and final
+    semicolon from sql_query for use with
+    sqlparse package
+
+    Args:
+        sql_query (str): The raw SQL query
+
+    Returns:
+        str: The cleaned SQL query
+    """
+    return " ".join(sql_query.splitlines()).strip().rstrip(";")
+
+
+def replace_temp_database_name_reference(sql_query: str, database_name: str) -> str:
+    """
+    Replaces references to to the users temp database __temp__
+    with the database_name string provided.
+
+    Args:
+        sql_query (str): The raw SQL query as a string
+        database_name (str): The database name to replace __temp__
+
+    Returns:
+        str: The new SQL query which is sent to Athena
+    """
+    # check query is valid and clean
+    parsed = sqlparse.parse(clean_query(sql_query))
+    new_query = []
+    for query in parsed:
+        print(query)
+        check_temp_query(str(query))
+        for word in str(query).strip().split(" "):
+            if "__temp__." in word.lower():
+                word = word.lower().replace("__temp__.", f"{database_name}.")
+            new_query.append(word)
+        if ";" not in new_query[-1]:
+            last_entry = new_query[-1] + ";"
+        else:
+            last_entry = new_query[-1]
+        del new_query[-1]
+        new_query.append(last_entry)
+    return " ".join(new_query)
+
+
+def get_user_id_and_table_dir(
+    force_ec2: bool = False, region_name: str = "eu-west-1"
+) -> Tuple[str, str]:
+    if force_ec2:
+        provider = InstanceMetadataProvider(
+            iam_role_fetcher=InstanceMetadataFetcher(timeout=1000, num_attempts=2)
+        )
+        creds = provider.load().get_frozen_credentials()
+        sts_client = boto3.client(
+            "sts",
+            region_name=region_name,
+            aws_access_key_id=creds.access_key,
+            aws_secret_access_key=creds.secret_key,
+            aws_session_token=creds.token,
+        )
+    else:
+        sts_client = boto3.client("sts", region_name=region_name)
+
+    sts_resp = sts_client.get_caller_identity()
+    out_path = os.path.join("s3://", bucket, sts_resp["UserId"])
+    if out_path[-1] != "/":
+        out_path += "/"
+
+    return (sts_resp["UserId"], out_path)
+
+
+def get_database_name_from_userid(user_id: str) -> str:
+    unique_db_name = user_id.split(":")[-1].split("-", 1)[-1].replace("-", "_")
+    unique_db_name = temp_database_name_prefix + unique_db_name
+    return unique_db_name
+
+
+def get_athena_client(force_ec2: bool = False, region_name: str = "eu-west-1"):
```

Comment:
This is largely repeated in get_table_id_and_table_dir - could that function use this one with some tweaks?

Reply:
Yeah I think you're right. I originally left separate because they're returning different clients but I am being lazy minded. Good catch! @gkelly900 I'll do this one.